### PR TITLE
Exit the polling loop if SIGTERM is detected

### LINKED ISSUE #547: SIGTERM no longer terminates the notary
Just prints, `run: zpoller_wait error`

### PR DESCRIPTION
**Pull request by weissjeffm**
Fixes #547

### CODE REVIEW

**Comment by toxeus** [comment]
@monetas-ci re-test this please

File: src/server/MessageProcessor.cpp [hunk 1]
```diff
@@ -202,6 +202,12 @@ void MessageProcessor::run()
             processSocket();
             continue;
         }
+        if (zpoller_terminated(zmqPoller_)) {
+            otErr << __FUNCTION__ << 
+                ": zpoller_terminated - process interrupted or parent context destroyed\n";
```
Comment (toxeus) [inline src/server/MessageProcessor.cpp]: Needs to be split:
```c++
": zpoller_terminated - process "
<< "interrupted or parent context destroyed\n";

**Comment by toxeus** [comment]
@monetas-ci re-test this please

File: src/server/MessageProcessor.cpp [hunk 2]
```diff
@@ -202,6 +202,13 @@ void MessageProcessor::run()
             processSocket();
             continue;
         }
+        if (zpoller_terminated(zmqPoller_)) {
+            otErr << __FUNCTION__ << 
+                ": zpoller_terminated - process interrupted or"
+                << " parent context destroyed\n";
```
Comment (toxeus) [inline src/server/MessageProcessor.cpp]: @weissjeffm sorry for the inconvenience but the formatter expects the following:
```c++
otErr << __FUNCTION__ 
      << ": zpoller_terminated - process interrupted or"
      << " parent context destroyed\n";
```

**Comment by weissjeffm** [comment]
@monetas-ci re-test this please

File: src/server/MessageProcessor.cpp [hunk 3]
```diff
@@ -202,6 +202,13 @@ void MessageProcessor::run()
             processSocket();
             continue;
         }
+        if (zpoller_terminated(zmqPoller_)) {
+            otErr << __FUNCTION__
```
Comment (toxeus) [inline src/server/MessageProcessor.cpp]: @weissjeffm It's a hassle when you don't have the formatter locally: you have a trailing space here. I'm sure that emacs can highlight them ;)

**Comment by toxeus** [comment]
@monetas-ci re-test this please

**Comment by toxeus** [comment]
Well done! :+1:

**Pull request merged**